app/services/datastore/application_rehydration.rb

```diff
@@ -36,11 +37,22 @@ def split_case?
       parent.return_details.reason.inquiry.split_case?
     end
 
+    # `is_means_tested` is not part of Schema, requires calculation
+    def means_tested
+      parent.means_passport.include?('on_not_means_tested') ? YesNoAnswer::NO : YesNoAnswer::YES
+    end
+
+    def client_has_partner
+      return if not_means_tested?
```

Comment:
client_has_partner will not have been asked, so will be `nil` . It's not in the datastore yet, so needs handling here

Reply:
same here could be `means_tested?`

Reply:
means_tested method is rebuilding the attribute called `is_means_tested` on the `crime_application` which is a YesNoAnswer

no_means_tested? is a predicate helper method used here to skip a local rehydration of partner that is not stored in the db and needs to be nil in this scenario